loopd/swapclient_server.go

```diff
@@ -409,16 +414,26 @@ func (s *swapClientServer) LoopIn(ctx context.Context,
 		}
 		req.LastHop = &lastHop
 	}
-	hash, htlc, err := s.impl.LoopIn(ctx, req)
+
+	hash, htlcP2WSH, htlcNP2WSH, err := s.impl.LoopIn(ctx, req)
 	if err != nil {
 		log.Errorf("Loop in: %v", err)
 		return nil, err
 	}
 
+	var legacyHtlcAddress string
+	if in.ExternalHtlc {
+		legacyHtlcAddress = htlcNP2WSH.String()
+	} else {
+		legacyHtlcAddress = htlcP2WSH.String()
```

Comment:
For loop in, I'd think the legacy address can always be set to NP2WSH, which it was before

Reply:
That'd mean that unless an external htlc is requested we'll always stay with np2wsh?

Reply:
Resolved offline. Missed that we are also switching over the internal address type